nupic/torch/modules/k_winners.py

```diff
@@ -271,21 +269,23 @@ def __init__(self, n, k, channels, kInferenceFactor=1.0, boostStrength=1.0,
     self.register_buffer("dutyCycle", torch.zeros((1, channels, 1, 1)))
 
 
-  def forward(self, x):
-    # Apply k-winner algorithm if k < n, otherwise default to standard RELU
-    if self.k >= self.n:
-      return F.relu(x)
+  def extra_repr(self):
+    return 'percent_on={}, boostStrength={}, dutyCyclePeriod={}'.format(
+      self.percent_on, self.boostStrength, self.dutyCyclePeriod)
 
-    if self.training:
-      k = self.k
-    else:
-      k = min(int(round(self.k * self.kInferenceFactor)), self.n)
 
-    x = k_winners2d.apply(x, self.dutyCycle, k, self.boostStrength)
+  def forward(self, x):
 
+    self.n = np.prod(x.shape[1:])
     if self.training:
+      k = int(round(self.n * self.percent_on))
+      x = k_winners2d.apply(x, self.dutyCycle, k, self.boostStrength)
       self.updateDutyCycle(x)
 
+    else:
+      k = int(round(self.n * self.percent_on_inference))
```

Comment:
Similar to previous comment on `self.n`,  both `k`s (inference and training) could be computed only once when we compute `n`